### ISSUE TITLE
Un used variable is emmitted while transpiling enums.

### ISSUE THREAD

**Issue by madhurakhal** [Microsoft/TypeScript#4620]
# TypeScript code
<pre>
<code>
module Workspace {
  export enum State {
    IN_PROGRESS,
    BUFFER_RENDERED,
    SKIPPED,
    TIME_OUT
  }
}
</code>
</pre>

# Compled JavaScript Code.
<pre>
 <code>
var Workspace;
(function (Workspace) {
    (function (State) {
        State[State["IN_PROGRESS"] = 0] = "IN_PROGRESS";
        State[State["BUFFER_RENDERED"] = 1] = "BUFFER_RENDERED";
        State[State["SKIPPED"] = 2] = "SKIPPED";
        State[State["TIME_OUT"] = 3] = "TIME_OUT";
    })(Workspace.State || (Workspace.State = {}));
    var State = Workspace.State; // this code is use no where.
})(Workspace || (Workspace = {}));
</code>
</pre>

I am using TypeScript version 1.5.3

**Comment by xLama** [comment]
It´s nccessary to use State enum. Consider this example:

´´´
module Workspace {
  export enum State {
    IN_PROGRESS,
    BUFFER_RENDERED,
    SKIPPED,
    TIME_OUT
  }
  
  class A{
	  buffer:State = State.BUFFER_RENDERED;
  }
  
}
´´´
JAVASCRIPT

```
var Workspace;
(function (Workspace) {
    (function (State) {
        State[State["IN_PROGRESS"] = 0] = "IN_PROGRESS";
        State[State["BUFFER_RENDERED"] = 1] = "BUFFER_RENDERED";
        State[State["SKIPPED"] = 2] = "SKIPPED";
        State[State["TIME_OUT"] = 3] = "TIME_OUT";
    })(Workspace.State || (Workspace.State = {}));
    var State = Workspace.State;
    var A = (function () {
        function A() {
            this.buffer = State.BUFFER_RENDERED;
        }
        return A;
    })();
})(Workspace || (Workspace = {}));
```

**Comment by madhurakhal** [comment]
Yep your code looks perfectly fine and reasonable when using the State enum at the same file. But In my case I am using State enum in different files. which means the only enum related code is there. So
```
var State = Workspace.State;
```
is unused one and also I am using Visual Studio Code which also complain the same.

#### Use case seniorio in another file after importing the enum (I am using backbone model)
```
this.model.set('state', Workspace.State.BUFFER_RENDERED);
```

**Comment by DanielRosenwasser** [comment]
@madhurakhal I don't think removing unused declarations is something we're aiming to achieve - a namespace is certainly consumable by others if any other code occurs after this. This is an optimization that linters and/or optimizers could work on.

_________________

@xLama 

> JS could be
> ```JavaScript
var Workspace;
(function (Workspace) {
    var State;
    (function (State) {
        State[State["IN_PROGRESS"] = 0] = "IN_PROGRESS";
        State[State["BUFFER_RENDERED"] = 1] = "BUFFER_RENDERED";
        State[State["SKIPPED"] = 2] = "SKIPPED";
        State[State["TIME_OUT"] = 3] = "TIME_OUT";
    })(State || (State = {}));
    Workspace.State = State;
})(Workspace || (Workspace = {}));
```

The idea point is that enums merge with prior declarations (hence the `Workspace.State || (Workspace.State = {})`.) In your example, you create a variable that is always going to start out undefined (`State`), then initialize it, and then overwrite `Worspace.State` which isn't the desired behavior.

```TypeScript
namespace E {
    export const C = 20;
}

enum E {
	A = 10,
	B = 20,
}
```

**Comment by RyanCavanaugh** [comment]
@rbuckton is this fixed in the transforms branch?

**Comment by RyanCavanaugh** [comment]
Accepting PRs for the transforms branch to make the emit the same as the namespace emit code (declare a var above the IIFE, then assign to the var in the IIFE arguments)
